### PR TITLE
migrate java/scala library to scala 2.12

### PR DESCRIPTION
**Pull request by shuttie**
It's a pity that this library is still unavailable for Scala 2.12. This PR tries to fix it.

Also fixes https://github.com/snowplow/referer-parser/issues/153

### CODE REVIEW

File: java-scala/project/plugins.sbt [hunk 1]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.3")
```
Comment (BenFradet) [inline java-scala/project/plugins.sbt]: I think we can do without this plugin?

File: java-scala/build.sbt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+organization  := "com.snowplowanalytics"
+version       := "0.3.0"
+description   := "Library for extracting marketing attribution data from referer URLs"
+scalaVersion  := "2.12.4"
+crossScalaVersions := Seq("2.11.11", "2.12.4")
+scalacOptions := Seq("-deprecation", "-encoding", "utf8")
+resolvers     += "SnowPlow Analytics Maven repo" at "http://maven.snplow.com/releases/"
```
Comment (BenFradet) [inline java-scala/build.sbt]: the resolver isn't required I think

File: java-scala/.gitignore [hunk 3]
```diff
@@ -10,3 +10,7 @@ lib_managed/
 src_managed/
 project/boot/
 project/plugins/project/
+.idea
+old
```
Comment (BenFradet) [inline java-scala/.gitignore]: can you remove old and .idea? these seem to be specific to you

File: java-scala/build.sbt [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+libraryDependencies ++= Seq(
+  "org.yaml" % "snakeyaml" % "1.19",
+  "org.apache.httpcomponents"  %  "httpclient"           % "4.5.3",
+  "org.specs2"                 %% "specs2"               % "2.5"       % "test",
+  "junit"                      % "junit"                 % "4.12"       % "test",
+  "org.json"                   % "json"                  % "20170516"        % "test",
+  "org.scalacheck"             %% "scalacheck"           % "1.13.5"  % "test",
+  "org.json4s"                %% "json4s-jackson"        % "3.5.3"      % "test",
+  "org.json4s"                %% "json4s-scalaz"         % "3.5.3"      % "test"
```
Comment (BenFradet) [inline java-scala/build.sbt]: could we align those dependencies?

**Comment by shuttie** [comment]
I've also signed CLA, but the badge `cla-signed` is still not here :(

**Comment by snowplowcla** [comment]
@shuttie has signed the Individual Contributor License Agreement. Thanks so much

**Pull request merged**